src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts

```diff
@@ -362,15 +365,20 @@ registry.registerWorkbenchAction(SyncActionDescriptor.create(QuickOpenTermAction
 const actionBarRegistry = Registry.as<IActionBarRegistry>(ActionBarExtensions.Actionbar);
 actionBarRegistry.registerActionBarContributor(Scope.VIEWER, QuickOpenActionTermContributor);
 
-(<panel.PanelRegistry>Registry.as(panel.Extensions.Panels)).registerPanel(panel.PanelDescriptor.create(
-	TerminalPanel,
-	TERMINAL_PANEL_ID,
-	nls.localize('terminal', "Terminal"),
-	'terminal',
-	40,
-	TERMINAL_COMMAND_ID.TOGGLE
-));
-Registry.as<panel.PanelRegistry>(panel.Extensions.Panels).setDefaultPanelId(TERMINAL_PANEL_ID);
+const VIEW_CONTAINER = Registry.as<IViewContainersRegistry>(ViewContainerExtensions.ViewContainersRegistry).registerViewContainer({
+	id: TERMINAL_VIEW_ID,
+	name: nls.localize('terminal', "Terminal"),
+	ctorDescriptor: new SyncDescriptor(ViewPaneContainer, [TERMINAL_VIEW_ID, TERMINAL_VIEW_ID, { mergeViewWithContainerWhenSingleView: true, donotShowContainerTitleWhenMergedWithContainer: true }]),
+	focusCommand: { id: TERMINAL_COMMAND_ID.FOCUS }
+}, ViewContainerLocation.Panel);
+Registry.as<panel.PanelRegistry>(panel.Extensions.Panels).setDefaultPanelId(TERMINAL_VIEW_ID);
+
+Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews([{
+	id: TERMINAL_VIEW_ID,
+	name: nls.localize('terminal', "Terminal"),
+	canToggleVisibility: false,
```

Comment:
No idea what this is meant to drive, I can still right click to toggle the visibility in the panel?

Reply:
This shall be set to `true` unless we do not want terminal to hidden.